### PR TITLE
Update image to extract java ca certs

### LINKED ISSUE #342: Build fails as Debian 8 jessie-updates disappeared
http://httpredir.debian.org/debian/dists/jessie-updates/ is gone. (404)

```
$ docker pull gcr.io/google_appengine/debian8
$ docker run --rm gcr.io/google_appengine/debian8 apt-get update

Ign http://httpredir.debian.org jessie InRelease
Ign http://httpredir.debian.org jessie-updates InRelease
Get:1 http://security.debian.org jessie/updates InRelease [44.9 kB]
Get:2 http://httpredir.debian.org jessie Release.gpg [2420 B]
Ign http://httpredir.debian.org jessie-updates Release.gpg
Get:3 http://httpredir.debian.org jessie Release [148 kB]
Ign http://httpredir.debian.org jessie-updates Release
Err http://httpredir.debian.org jessie-updates/main amd64 Packages
  
Err http://httpredir.debian.org jessie-updates/main amd64 Packages
  
Err http://httpredir.debian.org jessie-updates/main amd64 Packages
  
Err http://httpredir.debian.org jessie-updates/main amd64 Packages
  
Err http://httpredir.debian.org jessie-updates/main amd64 Packages
  404  Not Found
Get:4 http://httpredir.debian.org jessie/main amd64 Packages [9098 kB]
Get:5 http://security.debian.org jessie/updates/main amd64 Packages [824 kB]
Fetched 10.1 MB in 25s (397 kB/s)
W: Failed to fetch http://httpredir.debian.org/debian/dists/jessie-updates/main/binary-amd64/Packages  404  Not Found

E: Some index files failed to download. They have been ignored, or old ones used instead.
```

As a result, `cacerts_java` fails.

```
[589 / 630] 9 / 22 tests; SkylarkAction java/cacerts_java.tar; 34s linux-sandbox ... (3 actions running)
ERROR: /usr/local/google/home/chanseok/workspace4E/distroless-orig/java/BUILD:9:1: SkylarkAction java/cacerts_java.tar failed (Exit 100) cacerts.build failed: error executing command bazel-out/k8-fastbuild/bin/java/cacerts.build

Use --sandbox_debug to see verbose messages from the sandbox
+ docker load -i bazel-out/k8-fastbuild/bin/external/debian8/image/image.tar
++ docker run -d bazel/image:image sh -c 'apt-get update && apt-get install -y -q ca-certificates-java'
+ cid=67669ce6bce77d7d48c4d375e0ef852230657f173ff3a6097bf448c8ace2b5fa
+ docker attach 67669ce6bce77d7d48c4d375e0ef852230657f173ff3a6097bf448c8ace2b5fa
W: Failed to fetch http://httpredir.debian.org/debian/dists/jessie-updates/main/binary-amd64/Packages  404  Not Found

E: Some index files failed to download. They have been ignored, or old ones used instead.
Loaded image: bazel/image:image
Ign http://httpredir.debian.org jessie InRelease
Ign http://httpredir.debian.org jessie-updates InRelease
Get:1 http://security.debian.org jessie/updates InRelease [44.9 kB]
Get:2 http://httpredir.debian.org jessie Release.gpg [2420 B]
Ign http://httpredir.debian.org jessie-updates Release.gpg
Get:3 http://httpredir.debian.org jessie Release [148 kB]
Ign http://httpredir.debian.org jessie-updates Release
Err http://httpredir.debian.org jessie-updates/main amd64 Packages

Err http://httpredir.debian.org jessie-updates/main amd64 Packages

Err http://httpredir.debian.org jessie-updates/main amd64 Packages

Err http://httpredir.debian.org jessie-updates/main amd64 Packages

Err http://httpredir.debian.org jessie-updates/main amd64 Packages
  404  Not Found
Get:4 http://httpredir.debian.org jessie/main amd64 Packages [9098 kB]
Get:5 http://security.debian.org jessie/updates/main amd64 Packages [824 kB]
Fetched 10.1 MB in 25s (395 kB/s)
INFO: Elapsed time: 42.329s, Critical Path: 41.58s
INFO: 16 processes: 16 linux-sandbox.
FAILED: Build did NOT complete successfully
```

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #342.

Note we have tests to connect to www.google.com, such as [this](https://github.com/GoogleContainerTools/distroless/blob/master/java/testdata/java8_certs.yaml).

### CODE REVIEW

**Comment by chanseokoh** [comment]
@GoogleContainerTools/java-tools-build I got a confirmation that we should stop using `gcr.io/google_appengine/debian8` and migrate to `gcr.io/google_appengine/debian9`.

File: cacerts/java.bzl [hunk 1]
```diff
@@ -27,7 +27,8 @@ def _impl(ctx):
 set -ex
 docker load -i {0}
 # Install the certs in the builder image.
-cid=$(docker run -d {1} sh -c "apt-get update && apt-get install -y -q ca-certificates-java")
+# ln: the default non-interactive shell is dash, which interferes with the post install script.
+cid=$(docker run -d {1} sh -c "ln -sf bash /bin/sh && apt-get update && apt-get install -y -q ca-certificates-java")
```
Comment (briandealwis) [inline cacerts/java.bzl]: This is odd as from poking in the .deb, the `postinst` file has a `#!/bin/bash` as its shebang.

Comment (briandealwis) [reply cacerts/java.bzl]: Oh, maybe it's one of the prereqs.

Comment (chanseokoh) [reply cacerts/java.bzl]: It uses `!#/bin/sh`, but it has `==` (double equals) in a conditional that bash tolerates but dash doesn't.

File: .travis.yml [hunk 2]
```diff
@@ -17,9 +17,8 @@ jdk:
 install:
   - export PATH=$PATH:$HOME/bin && mkdir -p $HOME/bin
   - wget https://github.com/bazelbuild/buildtools/releases/download/0.22.0/buildifier && chmod +x buildifier && mv buildifier $HOME/bin/
-  - wget https://github.com/bazelbuild/bazel/releases/download/0.19.2/bazel-0.19.2-linux-x86_64 && mv bazel-0.19.2-linux-x86_64 bazel && chmod +x bazel && mv bazel $HOME/bin/
+  - wget https://github.com/bazelbuild/bazel/releases/download/0.24.0/bazel-0.24.0-linux-x86_64 && mv bazel-0.24.0-linux-x86_64 bazel && chmod +x bazel && mv bazel $HOME/bin/
```
[the remark this answers is not in the document]
Comment (chanseokoh) [reply .travis.yml]: Not required, but I will need this soon to use latest docker bazel rules.

**Comment by chanseokoh** [comment]
No idea about `openjdk`. Not even sure why we are using `gcr.io/google_appengine/debian`, as I think we just download the `ca-certificates-java` package from the public Debian repository. I thought I'd rather not change the behavior until we understand the details.

File: cacerts/java.bzl (hunk 1 above)
Comment (chanseokoh) [reply cacerts/java.bzl]: I mean, there may be a script with the bash shebang, but that is not the script causing an issue.

Comment (chanseokoh) [reply cacerts/java.bzl]: The error we see is from the `jks-keystore` script at line 56.

```
Running hooks in /etc/ca-certificates/update.d...

/etc/ca-certificates/update.d/jks-keystore: 56: [: amd64: unexpected operator
done.
```

The version of the package being installed is 20170929~deb9u1.

```
# dpkg -l | grep certif
ii  ca-certificates              20161130+nmu1+deb9u1  all          Common CA certificates
ii  ca-certificates-java         20170929~deb9u1       all          Common CA certificates (JKS keystore)
```

`jks-keystore.hook.in` from the matching date 20170929 has `!#/bin/sh` and uses `==` at line 56: https://salsa.debian.org/java-team/ca-certificates-java/blob/debian/20170929/debian/jks-keystore.hook.in#L56

**Pull request merged**